### ISSUE TITLE
Tasks to add sidecar containers in NSM

### ISSUE THREAD

**Issue by prateekgogia** [networkservicemesh/networkservicemesh#119]
These are the high levels tasks we need to do for adding sidecar containers.

- [ ] Create side car container image

- [x] Add side car container spec 

- [x] Create MutatingWebhookConfiguration for API server

- [x] Create MutatingAdmissionWebhook server for handling admission requests

- [x] MutatingAdmissionWebhook service and pod spec

- [x] Injector webhook configmap, containing container spec information to be injected in target pod

- [x] TLS authentication between API server and MutatingAdmissionWebhook server

**Comment by lixuna** [comment]
7/27/2018: blocked, needs cert approved

Related PR https://github.com/ligato/networkservicemesh/pull/120

**Comment by prateekgogia** [comment]
Need to enable CI for this, so will keep this open for now.

**Comment by mestery** [comment]
This should be unblocked now as we have gotten rid of Minikube at this point. Travis-ci now uses kubeadm-dind-cluster (which uses kubeadm to spin-up the cluster), and circle-ci uses cross-cloud-ci, which also uses kubeadm.

**Comment by prateekgogia** [comment]
Sure, I will revisit it this week. Thanks!

**Comment by lixuna** [comment]
Moving to deferred

**Comment by nickolaev** [comment]
Is this still on the table? Isn't the basic [NSC](https://github.com/ligato/networkservicemesh/tree/master/examples/cmd/nsc) good enough as a "init container"?
